bool objR
bool objT
bool objL
